File: campaign_manager/gos/serializers/npc.py
from gos.models.npc import NPC
from gos.serializers.npc_class import NPCClassSerializer
from gos.serializers.spell_slot import SpellSlotSerializer
from rest_framework import serializers

class NpcDetailSerializer(serializers.ModelSerializer):
    spell_slots = SpellSlotSerializer(many=True)
    npc_class = NPCClassSerializer()

    class Meta:
        model = NPC
        fields = [
            'id',
            'name',
            'bio',
            'race',
            'alignment',
            'personality_traits',
            'ideals',
            'bonds',
            'flaws',
            'notes',
            'secrets',
            'npc_class',
            'current_hp',
            'spell_slots'
        ]

class NpcListSerializer(serializers.ModelSerializer):
    class Meta:
        model = NPC
        fields = ['id', 'name']

File: src/app/class/npc.ts
import { NpcClass } from './npc-class';

export class Npc {
    id: number;
    name: string;
    bio: string;
    race: string;
    alignment: string
    personalityTraits: string;
    ideals: string;
    bonds: string;
    flaws: string;
    notes: string;
    secrets: string;
    npcClass: NpcClass;
    currentHp: number;

    constructor(props: any) {
        this.id = props.id;
        this.name = props.name;
        this.bio = props.bio;
        this.race = props.race;
        this.alignment = props.alignment;
        this.personalityTraits = props.personalityTraits;
        this.ideals = props.ideals;
        this.bonds = props.bonds;
        this.flaws = props.flaws;
        this.notes = props.notes;
        this.secrets = props.secrets;
        this.npcClass = props.npcClass;
        this.currentHp = props.currentHp;
    }
}
